File: app/src/main/java/com/switkows/mileage/widgets/StatisticsView.java
package com.switkows.mileage.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;
import android.widget.TextView;

/**
* Created by Trevor on 1/1/2015.
*/
public class StatisticsView extends RelativeLayout {

   private TextView mLabel;
   private TextView mValue;

   public StatisticsView(Context context, AttributeSet set) {
      super(context, set);
   }

   @Override
   public void onFinishInflate() {
      super.onFinishInflate();
      mLabel = (TextView)findViewById(android.R.id.text1);
      mValue = (TextView)findViewById(android.R.id.text2);
   }

   public void setLabel(String label) {
      mLabel.setText(label);
   }

   public void setValue(float value, String units) {
      setValue(String.format("%.1f %s", value, units));
   }

   public void setValuePercent(float value, String units) {
      setValue(String.format("%.2f%s", value, units));
   }

   public void setValue(String str) {
      mValue.setText(str);
   }
}

File: app/src/main/java/com/switkows/mileage/widgets/StatisticsView.kt
package com.switkows.mileage.widgets

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import android.widget.TextView

/**
 * Created by Trevor on 1/1/2015.
 */
class StatisticsView(context: Context, set: AttributeSet) : RelativeLayout(context, set) {

    private lateinit var mLabel: TextView
    private lateinit var mValue: TextView

    public override fun onFinishInflate() {
        super.onFinishInflate()
        mLabel = findViewById<TextView>(android.R.id.text1)
        mValue = findViewById<TextView>(android.R.id.text2)
    }

    fun setLabel(label: String) {
        mLabel.text = label
    }

    fun setValue(str: String) {
        mValue.text = str
    }
}
